src/ripple/beast/net/IPEndpoint.h

```diff
@@ -104,8 +106,13 @@ class Endpoint
     }
 
 private:
+    friend class ripple::OverlayImpl;
+    /** Updates the port */
+    void set_port (Port port) const
+    { m_port = port; }
+
     Address m_addr;
-    Port m_port;
+    mutable Port m_port;
```

Comment:
How about something like this to avoid setting `m_port` to `mutable`? https://github.com/HowardHinnant/rippled/commit/f31addfbe5fce0c83e309b26e209437b1705bab6

Reply:
I understand why you (@gregtatcam) made this change. We save copying about 20 bytes per address in `OverlayImpl`. However, to do this we had to:

1) Make `OverlayImpl` a friend. This makes code harder to audit, as I can't assume `OverlayImpl` only uses a public interface. 
2) Make a `set_port(...) const`. It's unusual to have a `set` function that is also `const`
3) Make the `port` member `mutable`. Usually, when something is mutable it's a mutex or cache or somesuch. It's unusual to make a "regular" data member mutable.

I saw @HowardHinnant's patch that solves some of these problems (and I like it). But it still leaves us with a friend function and makes the `addresses` parameter in/out. Those aren't killer, but I'm not sure it's worth it to save copying 20 bytes. I'd probably take the (small) performance hit in the copy rather than change the interfaces.